lms/djangoapps/courseware/module_render.py

```diff
@@ -554,7 +554,14 @@ def publish(block, event_type, event):
         if event_type == 'grade' and not is_masquerading_as_specific_student(user, course_id):
             handle_grade_event(block, event_type, event)
         else:
-            track_function(event_type, event)
+            aside_context = {}
+            for aside in block.runtime.get_asides(block):
+                if hasattr(aside, 'get_event_context'):
+                    aside_event_info = aside.get_event_context(event_type, event)
+                    if isinstance(aside_event_info, dict):
+                        aside_context.update(aside_event_info)
```

Comment:
@StraNNiKK - the intent was to store the output from each aside under a framework-provided key in order to separate them. We don't want asides defining data with the same keys overwriting one another. Something like this:

```json
{
    "context": {
        "asides": {
            "tagging": {
                "difficulty": "hard",
                "outcome": "multiplication"
            },
            "social": {
                "visible": ["facebook", "google_plus", "twitter"]
            }
        }
    }
}
```

Where "tagging" and "social" uniquely identify their respective asides.